OracleDatabase/19.3.0/.env

```diff
@@ -0,0 +1,53 @@
+# Oracle Database 19.3.0 configuration file
```

Comment:
Is there a way to inherit environment variables from another file?
A lot of these variable values are the same between 12.1.0.2 and 19.3.
Would be great if there is a way to have "global" defaults that can be overwritten per VM when needed.

Reply:
I'm not sure. I'll look into it, but doing it that way might make it more difficult for people to customize the boxes. It seems like people would either need to create two `.env.local` files at different levels, or combine two `.env` files into a single `.env.local` file. There's some benefit in keeping the boxes atomic.

Reply:
After some thought and experimentation, I recommend keeping the configuration parameters for each box separate. Using inheritance would make configuration more difficult.

Users should be able to modify any/all parameters for each box without affecting any other box. They should be able to do that in one place, without marking the Git tree as changed. In addition, they should be able to run multiple boxes (or multiple copies of the same box) simultaneously.

A `.env` file with common parameters would need to be in a separate directory, for example `vagrant-boxes/OracleDatabase/common`. Then the `.env` file for each box would contain only the parameters unique to that box.

If users wanted to modify both common parameters and box-specific parameters, they would have to copy parameters from both the common `.env` file and the box-specific `.env` file into a single `.env.local` file. This is a lot less convenient (and more error-prone) than just copying `.env` to `.env.local`.

A couple of other considerations also favor keeping the configurations separate:

- Many of the common parameters wouldn't apply to the XE boxes, which could be confusing.
- The common parameters may not always be common. For example, it might be a good idea to increase the default VM memory for the 18c and 19c boxes to 3 GB, to avoid memory warnings during database setup.

I understand the appeal of inheritance, but I think it'd be better to keep the boxes self-contained.

Reply:
That makes sense, thank you! I just wanted to make sure that we aren't missing a simple trick if there was one.